src/status_im2/contexts/chat/messages/content/view.cljs

```diff
@@ -0,0 +1,107 @@
+  [rn/view {:padding-top 2 :width 32}
+   (when (or (and (seq response-to) quoted-message) last-in-group? pinned)
+     (let [display-name (first (rf/sub [:contacts/contact-two-names-by-identity from]))
+           contact (rf/sub [:contacts/contact-by-address from])
+           photo-path (when-not (empty? (:images contact)) (rf/sub [:chats/photo-path from]))
+           online? (rf/sub [:visibility-status-updates/online? from])]
+       [quo/user-avatar {:full-name         display-name
+                         :profile-picture   photo-path
+                         :status-indicator? true
+                         :online?           online?
+                         :size              :small
+                         :ring?             false}]))])
+
+(defn author [{:keys [response-to last-in-group? pinned quoted-message from timestamp]}]
+  (when (or (and (seq response-to) quoted-message) last-in-group? pinned)
+    (let [display-name (first (rf/sub [:contacts/contact-two-names-by-identity from]))
+          {:keys [ens-verified added?]} (rf/sub [:contacts/contact-by-address from])]
+      [quo/author
+       {:profile-name   display-name
+        :chat-key       from
+        :time-str       (time/timestamp->time timestamp)
+        :contact?       added?
+        :verified?      ens-verified
+        ;; TODO not implemented yet
+        :nickname       nil
+        :ens-name       nil
+        :untrustworthy? nil}])))
+
+;;;; SYSTEM MESSAGES
+(defn system-message-content [{:keys [content-type quoted-message] :as message-data}]
+  (if quoted-message
+    [pin/system-message quoted-message] ;;TODO (flexsurfer) why we detect pin-message with quoted-message param?
+    (case content-type
+      ;; TEXT
+      constants/content-type-system-text [system.text/text-content message-data]
+      ;; COMMUNITY TODO (flexsurfer) refactor and move to status-im2
+      constants/content-type-community [old-message/community message-data]
+      ;; CONTACT REQUEST  TODO (flexsurfer) refactor and move to status-im2
+      constants/content-type-contact-request [old-message/system-contact-request message-data])))
+
+;;;; USER MESSAGES
+(defn user-message-content [{:keys [content-type] :as message-data}]
+  [rn/touchable-highlight {:underlay-color (colors/theme-colors colors/neutral-5 colors/neutral-90)
+                           :style          {:border-radius 16}
+                           :on-press       #()
+                           :on-long-press  #(rf/dispatch [:bottom-sheet/show-sheet
+                                                          {:content (drawers/reactions-and-actions message-data)}])}
+   [rn/view {:padding-horizontal 12 :padding-vertical 8 :flex-direction :row}
+    ;; AVATAR
+    [avatar message-data]
+    [rn/view {:margin-left 8}
+     ;; AUTHOR NAME
+     [author message-data]
+     (case content-type
+       ;; TEXT TODO (flexsurfer) refactor and move to status-im2
```

Comment:
@flexsurfer, I find these many TODO comments make the code rather confusing to read. You could group them into a single TODO saying they need to be refactored, or, better yet, just create an issue on Github.

Reply:
ok